src/main/java/org/apache/mesos/offer/ResourceUtils.java

```diff
@@ -211,6 +218,45 @@ public static ExecutorInfo clearResourceIds(ExecutorInfo executorInfo) {
             .build();
   }
 
+  public static boolean areDifferent(
+          ResourceSpecification oldResourceSpecification,
+          ResourceSpecification newResourceSpecification) {
+
+    Protos.Value oldValue = oldResourceSpecification.getValue();
+    Protos.Value newValue = newResourceSpecification.getValue();
+    if (!ValueUtils.equal(oldValue, newValue)) {
+      LOGGER.info("Values are different.");
+      return true;
+    }
+
+    if (!oldResourceSpecification.getRole().equals(newResourceSpecification.getRole())) {
```

Comment:
We should replace this with `Objects.equal()` call, as that's null safe like above `!ValueUtils.equal(oldValue, newValue)`